packages/integration-sdk-runtime/src/execution/dependencyGraph.ts

```diff
@@ -171,7 +171,7 @@ export function executeStepDependencyGraph<
         startTime,
         endTime,
         duration,
-        partialTypesToForce: partialTypesToForce,
+        partialTypes: existingResult.partialTypes.concat(partialTypes ?? []),
```

Comment:
When we assign the partial types, we should check they aren't already in the list so we don't duplicate them.